### PR TITLE
Separate chats and profiles

### PR DESCRIPTION
**Pull request by Vladyslav-Kuksiuk**
This PR fixes the problem described in [this comment](https://github.com/spine-examples/ChatSPN/pull/36#discussion_r1238767365).
Large `...Page` files became packages with smaller files.
The left sidebar moved to the `navigation` package with a model to make it work.  
`ChatsPage` became `ChatPage` and knows only about messages in one chat. 
Profiles UI moved to the `profile` package.
Common `@Composable` functions moved to the `component` package.

### CODE REVIEW

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatPage.kt [hunk 1]
```diff
@@ -860,106 +288,27 @@ private fun ChatDeletionDialog(
...
-    }
-}
-
 /**
  * Represents the list of messages in the chat.
  */
 @Composable
 private fun ChatMessages(model: ChatsPageModel) {
-    val selectedChat by model.selectedChat().collectAsState()
+    val chat = remember { model.chatState }
```
Comment (Artem-Semenov-dev) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatPage.kt:309]: This is a redundant variable. Please remove it.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/ProfilePage.kt [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+/**
+ * UI Model for the [ProfilePage].
+ */
+private class ProfilePageModel(
+    private val client: DesktopClient,
+    val userProfile: MutableState<UserProfile>,
+    val chatState: MutableState<ChatData?>,
+    val onBack: () -> Unit,
+    val openModal: (content: @Composable () -> Unit) -> Unit,
```
Comment (Artem-Semenov-dev) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/ProfilePage.kt:342]: It is not clear why we store the content of the modal window component in the model. Please explain this in the documentation.

**Review by Artem-Semenov-dev — COMMENTED** [review]
@Vladyslav-Kuksiuk please see my comments.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatPage.kt [hunk 3]
```diff
@@ -0,0 +1,1038 @@
...
+
+/**
+ * Settings to display the single message.
+ */
+private data class MessageDisplaySettings(
+    val color: Color,
+    val shape: Shape,
+    val alignment: Alignment,
+    val arrowWidth: Float
+)
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatPage.kt:1038]: This file is too long already. Let's split it by grouping related things by files.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/Navigation.kt [hunk 4]
```diff
@@ -26,48 +26,692 @@
...
-public class Navigation(private val client: DesktopClient) {
-    private val currentPage: MutableState<Page> = mutableStateOf(Page.REGISTRATION)
+@Composable
+private fun ChatNotChosenBox() {
+    Box(
+        Modifier.fillMaxSize(),
+        Alignment.Center
+    ) {
+        Text(
+            text = "Choose the chat",
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/Navigation.kt:436]: "a chat".

That's another point we should be thinking about.

Let's have a task to extract all text messages into resources. The code should be ready for internationalisation (or localisation, whichever you prefer) at any time.

File: client/src/main/kotlin/io/spine/examples/chatspn/desktop/Navigation.kt [hunk 5]
```diff
@@ -26,48 +26,692 @@
...
+ * @param client desktop client to find user profiles
+ */
+private fun MessagePreview.toMessageData(client: DesktopClient): MessageData {
+    return MessageData(
+        this.id,
+        client.findUser(this.user)!!,
+        this.content,
+        this.whenPosted
+    )
 }
```
Comment (armiol) [inline client/src/main/kotlin/io/spine/examples/chatspn/desktop/Navigation.kt:717]: Same as above. 717 lines is already too long. I wouldn't mind a file of this size, if it was much more homogeneous. But this one splits more or less logically into much more compact parts.

Please do.

**Review by Artem-Semenov-dev — COMMENTED** [review]
@Vladyslav-Kuksiuk LGTM in most, please see my comments.

**Review by armiol — APPROVED** [review]
@Vladyslav-Kuksiuk LGTM with several comments.

Also, I think, as the next minor step, we should group the extensions, so that public extensions to the same types reside in the same file, such as `FooBarExts.kt`.

**Pull request merged**